src/Framework/Sdk/SdkResultPathAndVersion.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// Encapsulates both the path to an SDK returned by an SDK resolver, and its version
+    /// </summary>
+    public sealed class SdkResultPathAndVersion
+    {
+        public string Path { get; }
+
+        public string Version { get; }
```

Comment:
I'm not totally clear on why version is interesting here. Can you elaborate in a doc comment?

Reply:
Before this change, the `SdkResult` had `Path` and `Version` properties, which were the path and optionally a version for the resolved Sdk.

Now that it can include multiple paths, I thought it made sense to support specifying a version for each path.  It would be a lot simpler if we didn't have this though, we could get rid of the whole `SdkResultPathAndVersion` type.

It looks like the versions returned in the SdkResult are only used for generating warnings such as:

- MSB4241: The SDK reference "{0}" version "{1}" was resolved to version "{2}" instead.  You could be using a different version than expected if you do not update the referenced version to match.
- MSB4240: Multiple versions of the same SDK "{0}" cannot be specified. The SDK version "{1}" already specified by "{2}" will be used and the version "{3}" will be ignored.

I would be in favor of saying that if you are returning multiple paths from an SDK resolver, you can't give a different version to each of them.  We only have one scenario for returning multiple paths and in that scenario we don't need to specify what version each path has.